jetstream/stream_config.go

```diff
@@ -26,154 +26,356 @@ import (
+		Template string `json:"template_owner,omitempty"`
 	}
 
-	// StreamSourceInfo shows information about an upstream stream source.
+	// StreamSourceInfo shows information about an upstream stream
+	// source/mirror.
 	StreamSourceInfo struct {
-		Name              string                   `json:"name"`
-		Lag               uint64                   `json:"lag"`
-		Active            time.Duration            `json:"active"`
-		FilterSubject     string                   `json:"filter_subject,omitempty"`
+		// Name is the name of the stream that is being replicated.
+		Name string `json:"name"`
+
+		// Lag informs how many messages behind the source/mirror operation is.
+		Lag uint64 `json:"lag"`
+
+		// Active informs when last the mirror or sourced stream had activity.
+		// Value will be -1 when there has been no activity.
+		Active time.Duration `json:"active"`
+
+		// FilterSubject is the subject filter defined for this source/mirror.
+		FilterSubject string `json:"filter_subject,omitempty"`
+
+		// SubjectTransforms is a list of subject transforms defined for this
+		// source/mirror.
 		SubjectTransforms []SubjectTransformConfig `json:"subject_transforms,omitempty"`
 	}
 
-	// StreamState is information about the given stream.
+	// StreamState is the state of a JetStream stream at the time of request.
 	StreamState struct {
-		Msgs        uint64            `json:"messages"`
-		Bytes       uint64            `json:"bytes"`
-		FirstSeq    uint64            `json:"first_seq"`
-		FirstTime   time.Time         `json:"first_ts"`
-		LastSeq     uint64            `json:"last_seq"`
-		LastTime    time.Time         `json:"last_ts"`
-		Consumers   int               `json:"consumer_count"`
-		Deleted     []uint64          `json:"deleted"`
-		NumDeleted  int               `json:"num_deleted"`
-		NumSubjects uint64            `json:"num_subjects"`
-		Subjects    map[string]uint64 `json:"subjects"`
+		// Msgs is the number of messages stored in the stream.
+		Msgs uint64 `json:"messages"`
+
+		// Bytes is the number of bytes stored in the stream.
+		Bytes uint64 `json:"bytes"`
+
+		// FirstSeq is the sequence number of the first message in the stream.
+		FirstSeq uint64 `json:"first_seq"`
+
+		// FirstTime is the timestamp of the first message in the stream.
+		FirstTime time.Time `json:"first_ts"`
+
+		// LastSeq is the sequence number of the last message in the stream.
+		LastSeq uint64 `json:"last_seq"`
+
+		// LastTime is the timestamp of the last message in the stream.
+		LastTime time.Time `json:"last_ts"`
+
+		// Consumers is the number of consumers on the stream.
+		Consumers int `json:"consumer_count"`
+
+		// Deleted is a list of sequence numbers that have been removed from the
+		// stream. This field will only be returned if the stream has been
+		// fetched with the DeletedDetails option.
+		Deleted []uint64 `json:"deleted"`
+
+		// NumDeleted is the number of messages that have been removed from the
+		// stream.
```

Comment:
is a bit more nuance than just `have been removed`.  It's about gaps so if the stream has 1,10,11 deleted is 9, but if the stream has 10,11 deleted is 0

Reply:
Yes, that's another weird tough thing to explain concisely...

Reply:
Changed to:
```go
		// NumDeleted is the number of messages that have been removed from the
		// stream. Only deleted messages causing a gap in stream sequence numbers
		// are counted. Messages deleted at the beginning or end of the stream
		// are not counted.
```